### PR TITLE
[Documentation]: ExampleApp improvements

### PR DESCRIPTION
**Pull request by AlicanAkyuz**
The type of this PR is: **Documentation**

This PR solves [WP-67]

### Description

This PR adds example test setup for the ExampleApp using RTL and improves the visibility of the app in our docs. Additionally, it aims to use palette v3 fully within the app. 

[WP-67]: https://artsyproduct.atlassian.net/browse/WP-67?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/v2/Apps/Example/README.md [hunk 1]
```diff
@@ -7,8 +7,8 @@ It includes:
 - [x] Routing
 - [x] Relay
 - [x] @principalField
-- [ ] Testing
-- [ ] Analytics
+- [x] Testing
+- [x] Analytics
```
Comment (AlicanAkyuz) [inline src/v2/Apps/Example/README.md:11]: marked this as done as there are examples of tracking in the app

File: src/v2/Apps/Example/__tests__/ExampleApp.jest.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import { screen } from "@testing-library/react"
+import { findMetaTagBySelector } from "v2/DevTools"
+
+jest.unmock("react-relay")
+
+const { renderWithRelay } = setupTestWrapperTL<ExampleApp_Test_Query>({
+  Component: props => {
+    return (
+      <MockBoot>
+        {/* @ts-expect-error PLEASE_FIX_ME_STRICT_NULL_CHECK_MIGRATION */}
```
Comment (dzucconi) [inline src/v2/Apps/Example/__tests__/ExampleApp.jest.tsx:15]: You'd just use `@ts-ignore` if you wanted to ignore this. This form is from the migration. To actually fix this problem you can just do something like `if (!props.system) return null` or whatever it's complaining about

Comment (dzucconi) [reply src/v2/Apps/Example/__tests__/ExampleApp.jest.tsx:15]: (Not that it matters a ton in this test code 🤷 )

Comment (AlicanAkyuz) [reply src/v2/Apps/Example/__tests__/ExampleApp.jest.tsx]: thanks for the hint 🙏

**Comment by damassi** [comment]
Thank you!

**Pull request merged**